cuda/matrix/coo_kernels.cu

```diff
@@ -321,11 +322,53 @@ void conj_transpose(std::shared_ptr<const CudaExecutor> exec,
 GKO_INSTANTIATE_FOR_EACH_VALUE_AND_INDEX_TYPE(
     GKO_DECLARE_COO_CONJ_TRANSPOSE_KERNEL);
 
+namespace kernel
+{
+
+template <typename ValueType>
+__global__ void initialize_zero_dense(size_type num_rows, size_type num_cols,
+		size_type stride, ValueType* result)
+{
+	auto tidx_x = threadIdx.x + blockDim.x * blockIdx.x;
+	auto tidx_y = threadIdx.y + blockDim.y * blockIdx.y;
+	if (tidx_x < num_rows && tidx_y < num_cols) {
+		result[tidx_x * stride + tidx_y] = zero<ValueType>();
+	}
+}
+
+template <typename ValueType, typename IndexType>
+__global__ __launch_bounds__(default_block_size) void fill_in_dense(size_type nnz, const IndexType* row_idxs,
+		const IndexType* col_idxs, const ValueType* values, size_type stride, ValueType* result)
+{
+	const auto tidx = threadIdx.x + blockDim.x * blockIdx.x;
+	if (tidx < nnz) {
+		result[stride * row_idxs[tidx] + col_idxs[tidx]] = values[tidx];
+	}
+}
+
+} // namespace kernel
+
 
 template <typename ValueType, typename IndexType>
 void convert_to_dense(
     std::shared_ptr<const CudaExecutor> exec, matrix::Dense<ValueType> *result,
-    const matrix::Coo<ValueType, IndexType> *source) NOT_IMPLEMENTED;
+    const matrix::Coo<ValueType, IndexType> *source)
+{
+	auto num_rows = result->get_size()[0];
+	auto num_cols = result->get_size()[1];
+	auto stride = result->get_stride();
+
+    auto nnz = source->get_num_stored_elements();
+
+    dim3 block_size = (32, 32, 1);
+    dim3 init_grid_dim = (ceildiv(num_rows, 32), ceildiv(num_cols, 32), 1);
```

Comment:
Also avoid using [magic numbers](https://en.wikipedia.org/wiki/Magic_number_(programming)).

Reply:
I just realized that you need to use `stride` instead of `num_cols` in case a stride was set to something bigger than `num_cols`.

Reply:
I don't think that's necessary since the content of the padding part of the dense matrix is not relevant, right?

Reply:
It is, because you calculate the index inside the threads like:
```c++
const auto column = global_index % stride;
const auto row = global_index / stride;
```
The stride can be used so each row start is assigned to the first thread in a warp, so there is always coalesced memory access. Therefore, it can increase performance by setting the stride to something bigger than `num_cols`, even though it increases the memory footprint.